### DISCUSSION TITLE
Different Cameras / EXIF Timezone interpretation

### DISCUSSION THREAD

**Discussion by deppchef** [LycheeOrg/Lychee#2471] [Q&A] [answered]
Hi all,

I stumbled over an issue in sorting when uploading images from two different cameras (in this case Nikon D750 and Z6 II) to the same album. I synched the cameras before the shooting (Time, Date, Daylight saving & Timezones) and the timely order when working in Lightroom is completely fine.

After export and upload to Lychee the images from the different cameras are being shifted 2 hours, which means the are more or less completely mixed up.
I expect this to be based on a different timezone data in the metadata of the Z6II. (after some research it looks like that timezones are not really reflected very good by EXIF / enhanced metadata). It never was an issue between D610 and D750 and D7000.

Long story short - Is there any option available to just ignore the timezone / daylight saving in Lychee? 
That would already solve my issue, as i take care of the times by myself. 

Alternatively - which timezone field in the metadata is used for sorting? maybe i could get rid of the respective values before export as well. 

I can still do a workaround by renaming the files and sorting by name, but that's not very highly sophisticated :)

Thanks,
Daniel

**Comment by ildyria** [comment]
Good point, If I am not wrong all the meta data are stored in the `taken_at` timestamp.
However there is also `taken_at_orig_tz` which contains the time zone of the shot.
Could you check what are the different values in those column in your DB ?

**Reply by deppchef** [reply]
Hi Benoit, 

Timestamp in "**taken_at**" seems to be perfectly fine for all cameras.
"**taken_at_orig_tz**" is set to UTC for the older Camera (D750) and to +02:00 for the newer (Z6 II).

Looks like the D750 (and all previous models) does not hand over the value in the metadata and Lychee takes the gallery default?  (was still set to UTC by default..)

CET plus Summertime means UTC+2 - which would be perfectly fine with the Z6II.
Just got the issue that there seems to be a difference in the way different generations of cameras treat the EXIF/IPTC fields.

![Screenshot 2024-06-18 203724](https://github.com/LycheeOrg/Lychee/assets/134516506/03514483-170c-474f-aff2-84ebef9782f2)

As stated above - this is no real showstopper, but a bit nasty when the gallery is mixed up although it is sorted :)
If you have any idea i'd be happy to know!

Thanks,
Daniel

**Comment by deppchef** [comment]
Hi Benoit, 

...just had a look at the EXIF/IPTC definitions. 
If my understanding is right, there are 3 relevant EXIF fields:
- DateTimeOriginal 
- SubSecTimeOriginal 
- OffsetTimeOriginal 

which are merged into IPTC field:
- Date Created

here's the Definition from IPTC consortium:
[link](https://iptc.org/std/photometadata/documentation/mappingguidelines/#exif-note-on-date-created
)

Seems like the older Camera does not reflect the "OffsetTimeOriginal" in a proper way.
Date Created for D750 looks like:
2024-06-08T13:02:18

where it shows this for the Z6II:
2024-06-08T13:03:40+02:00

The picture with the Z6 was taken later, but will be sorted before the other one, because of the time offset.

i now checked the Metadata of both files in detail and found out that the D750 filled the EXIF Field "OffsetTime" with +02:00 but NOT the field "OffsetTimeOriginal".
![image](https://github.com/LycheeOrg/Lychee/assets/134516506/fa00e5fb-a732-4ae1-90dd-c0aaba1ea5ae)

whereas the Z6 filled those:
![image](https://github.com/LycheeOrg/Lychee/assets/134516506/0ab8bd13-0c89-4696-afe9-d8dada299d70)


Not sure if you really want to deal with different interpretation of Metadata by type of camera :)

Regards,
Daniel

**Reply by deppchef** [reply]
last update.
i added "OffsetTimeOriginal" to a copy of a D750 picture - when setting to +02:00 like on the Z6II it behaves exactly the same after upload into Lychee. (used EXIF Pilot - works great but has no batch-option in the free version)

Maybe there is a possibility to fallback to "OffsetTime" if "OffsetTimeOriginal" is not available?
or an option to ignore the Offset completely for sorting.

**Reply by ildyria** [reply]
> https://github.com/LycheeOrg/php-exif

Probably something to have a look here.

**Reply by ildyria** [reply]
There are multiple problems at the same time to be honest:
- timezone of the picture
- timezone of the database
- timezone of the server

The problem we noticed is that those 3 are kinda inconsistent across database types: MySQL, PostgreSQL, SQLite3. This is why we had to use the offset field. :/

**Reply by deppchef** [reply]
no worries, i guess i found a decent workaround for the time being. 

I will just process all files through exiftool before the upload and either add or remove the tag "OffsetTimeOriginal"
just did a quick test and this works without problems - all files are either UTF or +2 hours now, depending on what i do.

i believe this whole thing is some kind of nightmare - different EXIF implementation, different DBs and last but not least different understandings... 

As i said - maybe adding an option to simply ignore the offset field for sorting would be fine already. 
you would just have to take care on the times for yourself 

thanks for the feedback and all your efforts! Lychee is really a great tool!

**Reply by ildyria** [reply, marked as answer]
https://github.com/LycheeOrg/php-exif/pull/72/files

That should help :)

**Reply by deppchef** [reply]
whoa... that was fast! 
As old-school ABAP-guy my php-knowledge is a bit limited... but from what i understand, this looks promising  👍 

Thanks!

**Reply by deppchef** [reply]
pulled the latest version and gave it a try. 
seems to work out fine!

**Reply by ildyria** [reply]
great to hear ! :)
